doc/sphinx/addendum/type-classes.rst

```diff
@@ -289,8 +302,27 @@ declares each projection as an instance.
 .. warn:: Ignored instance declaration for “@ident”: “@term” is not a class
 
    Using this ``:>`` syntax with a right-hand-side that is not itself a Class
-   has no effect (apart from emitting this warning). In particular, is does not
-   declare a coercion.
+   has no effect (apart from emitting this warning).
+
+.. warn:: 'Class >' currently does nothing, a coercion will be introduced in future versions.
+
+   :g:`Class >` will behave as :g:`Record >` in future versions,
+   declaring the construcotr a coercion. Remove the :g:`>` to avoid
+   this warning.
+
+.. warn:: 'Class >' is not supported for singleton typeclasses.
+
+   :g:`Record >` make the constructor a coercion, but there is no
+   constructor for singleton classes. Remove the :g:`>` to avoid this
+   warning.
+
+.. warn:: A coercion will be introduced in future versions when using ':>' in 'Class' declarations. Use '#[global] Existing Instance field.' instead if you don't want the coercion.
+
+   Currently, :g:`:>` only declares an instance. It will also declare
+   a coercion in future versions, like for any other record. Use
+   :cmd:`Existing Instance` or :cmd:`Coercion` separately rather than
+   :g:`:>` if you don't want both in the future (this will also avoid
+   this warning).
```

Comment:
Should be under the `Class` command description.  If the user wants both in the future, they have to change their now to make the warning go away?  That doesn't seem too friendly.

Reply:
Indeed, it's a bug (#2643 #9014), we are trying to sort it out in the less unfriendly way for most users. Feel free to express any suggestion that would improve that. Here is the status of the thing:
* this `:>` syntax in `Class` is used by about one third of the projects tested in the CI
* turning coercions on breaks four of them (c.f. the overlays above), mostly (for 3 out of the 4) because they actually need the coercion and declare it by hand which leads to an error "field is already a coercion" if we turn them on for the `:>`. For them, removing the `:>` as advised will be the only backward compatible fix.
* there seems to be a tiny portion of users (so, only 1 out of the more than 50 projects tested in the CI) for whom turning on coercions on `:>` in `Class` is harmful, that's the purpose of the warning so that they get a chance to hear about the change before it actually happens.